src/Expectation.php

```diff
@@ -90,4 +90,41 @@ public function toBePowerOf(int $number): PestExpectation
 
         return expect($this->value === $power)->toBeTrue();
     }
+
+    /**
+     * @return PestExpectation<TValue>
+     */
+    public function toBeFloat(): PestExpectation
+    {
+        $valueString = (string) $this->value;
+
+        $length = strlen($valueString);
+        $hasExponent = false;
+        $hasPrecision = false;
+
+        for ($i = 0; $i < $length; $i++) {
+            $char = $valueString[$i];
+
+            if ($char >= '0' && $char <= '9') {
+                continue;
+            }
+
+            if ($char === '.' && ! $hasPrecision && ! $hasExponent) {
+                $hasPrecision = true;
+
+                continue;
+            }
+
+            if (($char === 'e' || $char === 'E') && ! $hasExponent) {
+                $hasExponent = true;
+                $hasPrecision = true;
+
+                continue;
+            }
+
+            return expect(false)->toBeTrue();
+        }
+
+        return expect($hasPrecision)->toBeTrue();
```

Comment:
```suggestion
        return expect(is_float($this->value))->toBeTrue();
```

Reply:
Since we have php's `is_float()`, we can use it here.

Reply:
Well ``is_float`` was my initial guess, but I thought you didn't want to use built-in functions?

Reply:
No, I don't mind. It's always better to use built-in functions.